lib/pure/ropes.nim

```diff
@@ -15,6 +15,39 @@
 ## subtrees can be shared without copying.
 ## Leaves can be cached for better memory efficiency at the cost of
 ## runtime efficiency.
+##
+## A little picture makes everything clear
+## ::
+##|  "this string" & " is internally " & "represented as"
+##|            con  -- inner nodes do not contain raw data
+##|           /   \
+##|          /     \
+##|         /       \
+##|       con       "represented as"
+##|      /   \
+##|     /     \
+##|    /       \
+##|   /         \
+##|  /           \
+##|"this string"  " is internally "
+##|
+##|  Note that this is the same as:
+##|  "this string" & (" is internally " & "represented as")
+##|
+##|            con
+##|           /   \
+##|          /     \
+##|         /       \
+##|"this string"    con
+##|                /   \
+##|               /     \
+##|              /       \
+##|             /         \
+##|            /           \
+##|" is internally "        "represented as"
+##|
+##|  The 'con' operator is associative! This does not matter however for
+##|  the algorithms we use for ropes.
```

Comment:
This ASCII art is excessive and doesn't help much. Please remove it again. Also: Ropes are slower than mutable strings and we don't encourage anybody to use them. (In the Nim compiler they were a mistake.)